src/spn/algorithms/layerwise/clipper.py

```diff
@@ -0,0 +1,49 @@
+import torch
+import torch.nn.functional as F
+
+LOWER_BOUND = torch.tensor(1e-1)
+
+
+class DistributionClipper(object):
+    """
+    Clip distribution parameters.
+    """
+
+    def __init__(self, device, lower_bound=1e-5):
```

Comment:
Should we default the device argument as well? Is there a benefit to not doing it?

Reply:
I suppose we could default to `torch.device("cpu")` as is the default in PyTorch for the tensor class. I'll update that with an additional `to(device)` method to make it more PyTorch-y